tests/unit/test_local_project.py

```diff
@@ -0,0 +1,185 @@
+
+
+def test_local_project_repo_url():
+    """Get working_dir from git_repo"""
+    project = LocalProject(
+        git_repo=flexmock(active_branch="branch", working_dir="something"),
+        git_url=flexmock(),
+    )
+    assert project.git_repo
+    assert project.working_dir == "something"
+    assert project._branch == "branch"
+
+
+def test_local_project_repo():
+    """Get git_url from git_repo"""
+    project = LocalProject(
+        git_repo=flexmock(active_branch="branch", working_dir="something")
+        .should_receive("remote")
+        .replace_with(lambda: flexmock(urls=["git/url"]))
+        .once()
+        .mock()
+    )
+    assert project.git_repo
+    assert project.working_dir == "something"
+    assert project._branch == "branch"
+    assert project.git_url == "git/url"
+
+
+def test_clone_project_checkout_branch():
+    """Checkout existing branch"""
+    project = LocalProject(
+        git_repo=flexmock(
+            active_branch="branch",
+            working_dir="something",
+            branches={
+                "other": flexmock(checkout=lambda: None)
+                .should_receive("checkout")
+                .once()
+                .mock()
+            },
+        ),
+        branch="other",
+        git_url=flexmock(),
+    )
+    assert project.git_repo
+    assert project.working_dir == "something"
+    assert project._branch == "other"
+
+
+def test_clone_project_checkout_new_branch():
+    """Checkout newly created branch"""
+    branches = {}
+    project = LocalProject(
+        git_repo=flexmock(
+            active_branch="branch", working_dir="something", branches=branches
+        )
+        .should_receive("create_head")
+        .with_args("other")
+        .replace_with(
+            lambda x: branches.setdefault(
+                x, flexmock().should_receive("checkout").once().mock()
+            )
+        )
+        .once()
+        .mock(),
+        branch="other",
+        git_url=flexmock(),
+    )
+    assert project.git_repo
+    assert project.working_dir == "something"
+    assert project._branch == "other"
+
+
+def test_clone_project_service_repo_namespace():
+    """Get git_project from git_service and namespace/repo"""
+    project = LocalProject(
+        repo_name="repo",
+        namespace="namespace",
+        git_service=flexmock()
+        .should_receive("get_project")
+        .with_args(repo="repo", namespace="namespace")
+        .replace_with(lambda repo, namespace: flexmock())
+        .mock(),
+        git_url=flexmock(),
+        working_dir=flexmock(),
+        git_repo=flexmock(active_branch=flexmock()),
+    )
+    assert project.repo_name
+    assert project.namespace
+    assert project.git_service
+    assert project.git_project
+
+
+@mock.patch(
+    "packit.local_project.get_repo",
+    return_value=flexmock(working_dir="some/example/path", active_branch="branch"),
+)
+def test_local_project_clone(mock_get_repo):
```

Comment:
I believe you should be able to also use flexmock here to "patch" get_repo

Reply:
I'm just really trying to push for only a single mocking library, not two of them

Reply:
You are right, I've just seen that usage today in other repo, but have no time to correct that here.

I'll definitely vote for flexmock. It's super easy to use.

Reply:
Honestly, the upstream docs for flexmock is not too good, I often struggle with the proper usage.

My we should fix it upstream, or figure out best practices for us (and then send it upstream).